cpp/include/kvikio/shim/cufile_h_wrapper.hpp

```diff
@@ -25,11 +28,49 @@
 #ifdef KVIKIO_CUFILE_FOUND
 #include <cufile.h>
 #else
+
+// If cuFile isn't defined, we define some of the data types here.
+// Notice, this doesn't need to be ABI compatible with the cufile definitions.
+
+using CUfileHandle_t = void*;
+using CUfileOpError  = int;
+#define CUFILE_ERRSTR(x)          ("KvikIO not compiled with cuFile.h")
+#define CU_FILE_SUCCESS           0
+#define CU_FILE_CUDA_DRIVER_ERROR 1
+
+struct CUfileError_t {
+  CUfileOpError err;  // cufile error
+  CUresult cu_err;    // cuda driver error
+};
+
 using CUfileDriverControlFlags_t = enum CUfileDriverControlFlags {
-  CU_FILE_USE_POLL_MODE     = 0, /*!< use POLL mode. properties.use_poll_mode*/
-  CU_FILE_ALLOW_COMPAT_MODE = 1  /*!< allow COMPATIBILITY mode. properties.allow_compat_mode*/
+  CU_FILE_USE_POLL_MODE     = 0,
+  CU_FILE_ALLOW_COMPAT_MODE = 1
 };
-using CUfileHandle_t = void*;
+
+enum CUfileFileHandleType { CU_FILE_HANDLE_TYPE_OPAQUE_FD = 1 };
+
+struct CUfileDescr_t {
+  enum CUfileFileHandleType type;
+  struct handle_t {
+    int fd;
+  } handle;
+};
+
+const char* cufileop_status_error(...) { return "KvikIO not compiled with cuFile.h"; };
```

Comment:
```suggestion
static inline const char* cufileop_status_error(CUfileOpError err) { return CUFILE_ERRSTR(err); };
```
`static inline` for ODR, and `CUFILE_ERRSTR` for OAOO for the error string.